Added cartesio support

maukcc opened the pull request.

BagelOrb commented:
It's great to see community contributions to the fdm_material database. You're the first!

Hopefully we decide to merge this for the Cura 2.5 release. Sadly it's just a bit too late for the Cura 2.4 release.

maukcc commented:
Where do the setting key come from?
As they are not the same variables as the ones used in fdmextruder.
I need to add:
retraction_prime_speed
switch_extruder_prime_speed

awhiemstra commented:
The specification we originally wrote has a list, I do not know if we published that yet. In any case, the same list is reflected in cura/plugins/XmlMaterialProfile/XmlMaterialProfile.py line 581. These are currently, unfortunately, the only values allowed in the material profile. We have plans to change this, but we do not know when we will get around to it. As a workaround, what we do for the UM printers is to have the required settings in the quality profile.

Ghostkeeper commented on generic_petg.xml.fdm_material:
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Generic PLA profile. Serves as an example file, data in this file is not correct.
```
You should probably edit or remove this comment.

maukcc replied:
OK, edited this line.

Ghostkeeper commented on generic_petg.xml.fdm_material:
```diff
@@ -0,0 +1,38 @@
+            <material>PETG</material>
+            <color>Generic</color>
+        </name>
+        <GUID></GUID>
```
Currently we match a few things on this GUID. This empty GUID is currently unique but until we fix the dependency on unique GUIDs we should probably fill something in here.

maukcc replied:
The original author of these materials probably has a unique way of generating the numbers, So I do not want to fiddle with it.
Or we should make clear where to get a unique number.

awhiemstra replied:
I believe it is just a bit of Python code:

```
import uuid

uuid.uuid4()
```

or something along those lines.

Ghostkeeper replied:
I've generated a GUID for you:

    1cbfaeb3-1906-4b26-b2e7-6f777a8c197a

Since this is generic PETG, rather than a specific product, it doesn't matter what GUID there is as long as it is unique, since there are not going to be any spools with this GUID.

Ghostkeeper commented:
When you've made the changes you like, will you give us a ping here? A message? We'll look at your changes again then.

maukcc commented:
It`s pretty much done:

I changed the top lines to 
+Generic ABS profile. The data in this file may not be correct for your specific machine.

and added materials:
PETG
HIPS

The HIPS will need a ID number
both will need some additions for the UM printers

Ghostkeeper commented on generic_cpe.xml.fdm_material:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-Generic CPE profile. Serves as an example file, data in this file is not correct.
+Generic CPE profile. The data in this file may not be correct for your specific machine.
```
I like this change. Originally it was meant as an example file but this is not quite right any more. Thanks.

Ghostkeeper commented on generic_hips_xml.fdm_material:
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Generic ABS profile. The data in this file may not be correct for your specific machine.
```
This is the HIPS profile.

Ghostkeeper commented:
Some changes required. I made them for you. See here: https://github.com/maukcc/fdm_materials/pull/1

maukcc commented:
Thanks for making the changes!
